gateway/api_loader.go

```diff
@@ -599,6 +601,7 @@ func (d *DummyProxyHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 		var handler http.Handler
 		if r.URL.Hostname() == "self" {
+			httpctx.SetSelfLooping(r, true)
```

Comment:
The problem I have with this line is that I can comment it out and all tests pass.

Reply:
Because self-looping is not tested. 

The bug fix separates self-looping from looping to another API. If the request is self-looping, the auth_key middleware skips auth checks. Otherwise, it applies the business logic. We need to use the business logic of the auth key middleware if the request is not self-looping. The problem was skipping the auth-key middleware.